subsys/bluetooth/host/gatt.c

```diff
@@ -2781,11 +2781,11 @@ static void gatt_find_type_rsp(struct bt_conn *conn, uint8_t err,
 		value.end_handle = end_handle;
 		value.uuid = params->uuid;
 
-		attr.uuid = &uuid_svc.uuid;
-		attr.handle = start_handle;
-		attr.user_data = &value;
+		attr = (&(struct bt_gatt_attr)BT_GATT_ATTRIBUTE(
+			&uuid_svc.uuid, 0, NULL, NULL, &value));
+		attr->handle = start_handle;
```

Comment:
What's the advantage of doing this instead of passing a reference to a stack variable? You're just increasing stack usage with the extra pointer variable.

Reply:
I wanted to use the BT_GATT_ATTRIBUTE to initialize it. But need to have it together with the declaration.
I've moved it up now.

Reply:
You don't *need* to do it as part of the declaration, but if you don't you need an explicit `(struct bt_gatt_attr)` typecast in front. I think this begs the question whether we should include that typecast as part of the definition. That would also simplify places wanting to construct these pointers to anonymous variables.

Reply:
Thank you for mentioning that. Then I got what I wanted to do.

While including the cast would be useful I think we should do that separately and consistently.
We do have other init macros that should use the same pattern then in my opinion.